scripts/cat/status.py

```diff
@@ -0,0 +1,639 @@
+        """
+        # if we have a specific afterlife to send them to
+        if target:
+            self.add_to_group(
+                new_group=target,
+            )
+            return
+
+        # if we have an outsider who has never been a clancat, they go to the unknown residence
+        if self.is_outsider() and not self.is_former_clancat():
+            self.add_to_group(new_group=CatGroup.UNKNOWN_RESIDENCE)
+            return
+
+        # meanwhile clan cats go wherever their guide points them
+        if game.clan:
+            self.add_to_group(new_group=game.clan.instructor.status.group)
+        else:
+            self.add_to_group(new_group=CatGroup.STAR_CLAN)
+
+    def change_rank(self, new_rank: CatRank):
+        """
+        Changes the cats rank to the new_rank. Generally you shouldn't use just this to change a cat's rank!
+        cat.rank_change() should typically be called instead, since it will handle mentor switches and other complex
+        changes.
+        """
+        # checks that we don't add a duplicate group/rank pairing
+        if self.group_history:
+            last_entry = self.group_history[-1]
+            if last_entry["group"] == self.group and last_entry["rank"] == new_rank:
+                return
+
+        self.group_history.append(
+            {"group": self.group, "rank": new_rank, "moons_as": 0}
+        )
+
+    def change_group_nearness(self, group: CatGroup):
+        """
+        Flips the "near" bool of the given group.
+        """
+        for entry in self.standing_history:
+            if entry.get("group") == group:
+                if entry.get("near"):
+                    entry["near"] = False
+                else:
+                    entry["near"] = True
+
+    # RETRIEVE INFO
+    def get_standing_with_group(self, group: CatGroup) -> list[CatStanding]:
+        """
+        Returns the list of standings a cat has for the given group.
+        """
+        standing_list = []
+        for entry in self.standing_history:
+            if entry["group"] == group:
+                standing_list = entry["standing"]
+                break
+
+        return standing_list
+
+    def find_prior_clan_rank(self, clan: CatGroup = None):
+        """
+        Finds the last clan rank held of a current outsider
+        :param clan: pass the name of a clan to only return the cat's prior rank within that clan. Default is None, if
+        None then the last rank within any Clan will be returned.
+        """
+        if clan:
+            past_ranks = [
+                record["rank"]
+                for record in self.group_history
+                if record["group"] == clan
+            ]
+        else:
+            past_ranks = [
+                rank
+                for rank in self.all_ranks.keys()
+                if rank not in [CatRank.LONER, CatRank.KITTYPET, CatRank.ROGUE]
+            ]
+
+        return past_ranks[-1]
+
+    def in_player_clan(self) -> bool:
```

Comment:
```suggestion
    def alive_in_player_clan(self) -> bool:
```

As I mentioned elsewhere, I think this should explicitly mark that a cat is *alive* and part of the Clan. I know it's semantics, but to my mind a dead cat is still part of PlayerClan, they're just also in StarClan now. It would also make several statements more explicit, which is always good.